pkg/api/types.go

```diff
@@ -1889,6 +1889,8 @@ type EndpointAddress struct {
 	// Optional: Hostname of this endpoint
 	// Meant to be used by DNS servers etc.
 	Hostname string `json:"hostname,omitempty"`
+	// Optional: Node hosting this endpoint
+	NodeName string `json:"nodeName,omitempty"`
```

Comment:
optional fields should always be pointers.  I know that was not always enforced but it is true now.